frontend/src/views/SubmissionView.vue

```diff
@@ -42,28 +59,88 @@ async function onSubmit(data: any) {
 
 onMounted(async () => {
   submission.value = (await chefsService.getSubmission(props.formId, props.submissionId)).data;
+  documents.value = (await documentService.listDocuments(props.submissionId)).data;
 });
-
-const onUpload = async (file: File) => {
-  await documentService.createDocument(file, props.submissionId, getConfig.value.coms.bucketId);
-};
 </script>
 
 <template>
   <h1>Activity submission</h1>
 
-  <Button
-    class="mb-3"
-    @click="editable = !editable"
-  >
-    Edit
-  </Button>
-
-  <SubmissionForm
-    v-if="submission"
-    :editable="editable"
-    :submission="submission"
-    @cancel="onCancel"
-    @submit="onSubmit"
-  />
+  <TabView>
+    <TabPanel header="Info">
+      <Button
+        class="mb-3"
+        @click="editable = !editable"
+      >
+        Edit
+      </Button>
+
+      <SubmissionForm
+        v-if="submission"
+        :editable="editable"
+        :submission="submission"
+        @cancel="onCancel"
+        @submit="onSubmit"
+      />
+    </TabPanel>
+    <TabPanel header="Files">
+      <div class="grid border-1 surface-border border-round text-center">
+        <div class="col-2">
+          <FileUpload
+            :submission-id="props.submissionId"
+            @update:model-value="(e: Document) => documents.push(e)"
+          />
+        </div>
+        <div class="col-10 p-0">
+          <Carousel
+            v-if="documents.length"
+            :value="documents"
+            :num-visible="3"
+            :num-scroll="1"
+            :responsive-options="responsiveOptions"
+          >
```

Comment:
There's a chance we can reduce the amount of DOM element nesting by shifting the class grid stuff into the component itself with something like the following:
```vue
<FileUpload
  class="col-2"
  :submission-id="props.submissionId"
  @update:model-value="(e: Document) => documents.push(e)"
/>
<Carousel
  v-if="documents.length"
  class="col-10 p-0"
  :value="documents"
  :num-visible="3"
  :num-scroll="1"
  :responsive-options="responsiveOptions"
>
```

Reply:
Tried. Doesn't work in this instance. Something about extraneous non props being passed that can't be inherited.